### PR TITLE
Refactor/iso15118 charger interface

### PR DESCRIPTION
**Pull request by SebaLukas**
## Describe your changes
Changed all cmds and vars in iso15118_charger interface to `snake_case` and all types in iso15118_charger types to `PascalCase`. 

## Issue ticket number and link
In the iso15118_charger interface there is a wild mix between snake_case, camelCase & PascalCase. 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: interfaces/ISO15118_charger.yaml [hunk 1]
```diff
@@ -327,61 +327,19 @@ vars:
...
+    type: object
+    $ref: /iso15118_charger#/AppProtocols
+  v2g_messages:
     description: >-
       Debug - This element contains all V2G elements and should be used
       for debug purposes only
     type: object
-    $ref: /iso15118_charger#/V2G_Messages
-  Selected_Protocol:
+    $ref: /iso15118_charger#/V2gMessages
```
Comment (Pietfried) [inline interfaces/ISO15118_charger.yaml:341]: rather `V2GMessages` ?

File: modules/EvseManager/energy_grid/energyImpl.cpp [hunk 2]
```diff
@@ -425,81 +425,83 @@ void energyImpl::handle_enforce_limits(types::energy::EnforcedLimits& value) {
                     auto powersupply_capabilities = mod->get_powersupply_capabilities();
 
                     // tell car our new limits
-                    types::iso15118_charger::DC_EVSEMaximumLimits evseMaxLimits;
+                    types::iso15118_charger::DcEvseMaximumLimits evseMaxLimits;
```
Comment (Pietfried) [inline modules/EvseManager/energy_grid/energyImpl.cpp:428]: not sure if it makes sense to make this `evseMaxLimits` to snake case `evse_max_limits` as part of this PR as well

File: interfaces/ISO15118_charger.yaml [hunk 3]
```diff
@@ -175,149 +175,149 @@ cmds:
...
     type: "null"
-  currentDemand_Finished:
+  current_demand_finished:
     description: The charging process was finished
     type: "null"
   sae_bidi_mode_active:
       description: The SAE J2847 bidi mode is active
       type: "null"
-  EVCCIDD:
+  evccidd:
```
Comment (Pietfried) [inline interfaces/ISO15118_charger.yaml:213]: is it `evcc_idd` or `evcc_id` ?

**Review by Pietfried — APPROVED** [review]

File: interfaces/ISO15118_charger.yaml [hunk 4]
```diff
@@ -175,149 +175,149 @@ cmds:
...
       the MAC address of the EVCC in uppercase
     type: string
     pattern: ^[A-F0-9]{2}(:[A-F0-9]{2}){5}$
-  SelectedPaymentOption:
+  selected_payment_option:
     description: This element is used for indicating the payment type
     type: string
     $ref: /iso15118_charger#/PaymentOption
-  RequestedEnergyTransferMode:
+  requested_energy_transferMode:
```
Comment (corneliusclaussen) [inline interfaces/ISO15118_charger.yaml:223]: shouldn't it be transfer_mode?

File: interfaces/ISO15118_charger.yaml (hunk 1 above)
Comment (corneliusclaussen) [reply interfaces/ISO15118_charger.yaml:341]: V2gMessages looks weird but technically it is correct pascal case

**Review by corneliusclaussen — APPROVED** [review]
approved with some minor comments

File: interfaces/ISO15118_charger.yaml (hunk 1 above)
Comment (SebaLukas) [reply interfaces/ISO15118_charger.yaml:341]: I would prefer V2GMessages, but V2gMessages is correct here

File: modules/EvseManager/energy_grid/energyImpl.cpp (hunk 2 above)
Comment (SebaLukas) [reply modules/EvseManager/energy_grid/energyImpl.cpp:428]: Yes, that's right, I'll have a look if I can also find such variables

File: interfaces/ISO15118_charger.yaml (hunk 3 above)
Comment (SebaLukas) [reply interfaces/ISO15118_charger.yaml:213]: Thx :) `EVCCID` is correct. Then it should be evcc_id. I'll change that

File: interfaces/ISO15118_charger.yaml (hunk 4 above)
Comment (SebaLukas) [reply interfaces/ISO15118_charger.yaml:223]: Thx :) this is correct. I'll change that

File: modules/EvseManager/energy_grid/energyImpl.cpp (hunk 2 above)
Comment (SebaLukas) [reply modules/EvseManager/energy_grid/energyImpl.cpp]: Find some more. Hopefully all

**Pull request merged**